packages/payment-processor/test/payment/erc777-stream.test.ts

```diff
@@ -0,0 +1,201 @@
+  meta: {
+    transactionManagerMeta: {},
+  },
+  pending: null,
+  requestId: 'abcd',
+  state: RequestLogicTypes.STATE.CREATED,
+  timestamp: 0,
+  version: '1.0',
+};
+
+describe('erc777-stream', () => {
+  beforeEach(() => {
+    jest.restoreAllMocks();
+  });
+  describe('getRequestPaymentValues', () => {
+    it('handles ERC777', () => {
+      const values = getRequestPaymentValues(validRequest);
+      expect(values.expectedStartDate).toBe(expectedStartDate);
+      expect(values.expectedFlowRate).toBe(expectedFlowRate);
+      expect(values.paymentAddress).toBe(paymentAddress);
+      expect(values.paymentReference).toBe('86dfbccad783599a');
+    });
+  });
+
+  describe('encodePayErc20FeeRequest (used to pay and swap to pay)', () => {
+    it('should throw an error if the request is not erc777', async () => {
+      const request = Utils.deepCopy(validRequest) as ClientTypes.IRequestData;
+      request.currencyInfo.type = RequestLogicTypes.CURRENCY.ETH;
+
+      await expect(payErc777StreamRequest(request, wallet)).rejects.toThrowError(
+        'request cannot be processed, or is not an pn-erc777-stream request',
+      );
+    });
+
+    it('should throw an error if the currencyInfo has no value', async () => {
+      const request = Utils.deepCopy(validRequest);
+      request.currencyInfo.value = '';
+      await expect(payErc777StreamRequest(request, wallet)).rejects.toThrowError(
+        'request cannot be processed, or is not an pn-erc777-stream request',
+      );
+    });
+
+    it('should throw an error if currencyInfo has no network', async () => {
+      const request = Utils.deepCopy(validRequest);
+      request.currencyInfo.network = '';
+      await expect(payErc777StreamRequest(request, wallet)).rejects.toThrowError(
+        'request cannot be processed, or is not an pn-erc777-stream request',
+      );
+    });
+
+    it('should throw an error if request has no extension', async () => {
+      const request = Utils.deepCopy(validRequest);
+      request.extensions = [] as any;
+
+      await expect(payErc777StreamRequest(request, wallet)).rejects.toThrowError(
+        'Not a supported ERC777 payment network request',
+      );
+    });
+  });
+
+  describe.only('payErc777StreamRequest', () => {
+    it('should pay an ERC777 request with fees', async () => {
+      let tx;
+      let confirmedTx;
+      //initialize the superfluid framework...put custom and web3 only bc we are using ganache locally
+      const sf = await Framework.create({
+        networkName: 'custom',
+        provider,
+        dataMode: 'WEB3_ONLY',
+        resolverAddress: resolverAddress,
+        protocolReleaseVersion: 'test',
+      });
+
+      //use the framework to get the SuperToken
+      const daix = await sf.loadSuperToken('fDAIx');
+
+      //get the contract object for the erc20 token
+      const daiAddress = daix.underlyingToken.address;
+      const dai = new Contract(daiAddress, daiABI, wallet);
+
+      // //minting fDAI
```

Comment:
Please fix comment format + place

Reply:
sorry for the confusion in comments style